src/fstab-generator/fstab-generator.c

```diff
@@ -691,12 +692,29 @@ static int add_sysroot_mount(void) {
                 return 0;
         }
 
-        what = fstab_node_to_udev_node(arg_root_what);
-        if (!what)
-                return log_oom();
+        if (streq(arg_root_what, "tmpfs")) {
+                /* If root=tmpfs is specified, then take this as shortcut for a writable tmpfs mount as root */
+
+                what = strdup("rootfs"); /* just a pretty name, to show up in /proc/self/mountinfo */
+                if (!what)
+                        return log_oom();
+
+                fstype = arg_root_fstype ?: "tmpfs"; /* tmpfs, unless overriden */
```

Comment:
is it a valid use case to have root=tmpfs but then have something other than tmpfs as the rootfstype?

Reply:
well, IRL probably not. But dunno, people could use `root=tmpfs rootfstype=ramfs` if that's how they want to roll, i.e. get tmpfs-like behaviour, but then replace the fs implementation of tmpfs by ramfs, maybe to turn of swapping?

I mean, not sure this makes much sense, but if people explicitly specify an fstype in this case, i don't think we need to block them from doing so...

Reply:
sure, just checking if the combination made sense